hw/mcu/nordic/nrf52xxx/src/hal_gpio.c

```diff
@@ -394,7 +394,7 @@ hal_gpio_irq_init(int pin, hal_gpio_irq_handler_t handler, void *arg,
     hal_gpio_init_in(pin, pull);
 
 #if MYNEWT_VAL(MCU_GPIO_USE_PORT_EVENT)
-    conf = 0;
+    (void)conf;
```

Comment:
It could be possible to use if instead of #if then we would have syntax checking for 0 and 1 at once and possibly no need to (void) unused variables.
Not that I object to what has been done, just a thought for the future.